File: app/src/main/java/itc_android/com/itc_android/ui/palindrome/fragment/ResultFragment.java
package itc_android.com.itc_android.ui.palindrome.fragment;


import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;
import itc_android.com.itc_android.Constant;
import itc_android.com.itc_android.R;
import itc_android.com.itc_android.model.CPalindrome;


/**
 * A simple {@link Fragment} subclass.
 */
public class ResultFragment extends Fragment implements ResultAdapter.ListenerResultAdapter {

    @BindView(R.id.btnResult)
    Button btnResult ;
    @BindView(R.id.tvResult)
    TextView tvResult ;
    @BindView(R.id.rc_Item)
    RecyclerView recyclerView ;
    private ListenerResult listenerResult ;
    private ResultAdapter adapter ;
    private Unbinder unbinder;
    private List<CPalindrome> list ;
    private LinearLayoutManager llm ;

    public ResultFragment() {
        // Required empty public constructor
    }

    public void setListenerResult(ListenerResult listenerResult){
        this.listenerResult = listenerResult ;
        list = new ArrayList<>();

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view =  inflater.inflate(R.layout.fragment_result, container, false);
        unbinder = ButterKnife.bind(this,view);
        setupRecyclerView();
        return  view ;
    }

    public void setupRecyclerView() {
        llm = new LinearLayoutManager(getActivity(), LinearLayoutManager.VERTICAL, false);
        recyclerView.setLayoutManager(llm);
        adapter = new ResultAdapter(getActivity().getLayoutInflater(),getActivity(), this);
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onItemClicked(int position) {

    }

    @Override
    public void onResume() {
        super.onResume();
        Bundle bundle = new Bundle();
        bundle = getArguments();
        String value = bundle.getString(Constant.TAG_VALUE);
        String result = bundle.getString(Constant.TAG_RESULT);
        if (result.equals("true")){
            tvResult.setText(value +" : "+" is a palindrome");
        }
        else{
            tvResult.setText(value +" : "+" is not a palindrome");
        }
        list.add(new CPalindrome(value,result));
        Collections.reverse(list);
        adapter.setDataSource(list);
    }

    @OnClick(R.id.btnResult)
    public void onClickResult(){
        listenerResult.onResultAction();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        unbinder.unbind();
    }

    public interface ListenerResult {
        void onResultAction();
    }

}

File: app/src/main/java/itc_android/com/itc_android/ui/palindrome/fragment/ResultFragment.kt
package itc_android.com.itc_android.ui.palindrome.fragment


import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import java.util.ArrayList
import com.michaldrabik.kotlintest.utilities.extensions.dpToPx
import itc_android.com.itc_android.Constant
import itc_android.com.itc_android.R
import itc_android.com.itc_android.model.CPalindrome
import kotlinx.android.synthetic.main.fragment_result.view.*
import javax.inject.Inject


/**
 * A simple [Fragment] subclass.
 */
class ResultFragment : Fragment() {

    private var listenerResult: ListenerResult? = null
    @Inject lateinit var adapter: ResultAdapter
    private var list: MutableList<CPalindrome>? = null
    private var tvResult : TextView? = null

    fun setListenerResult(listenerResult: ListenerResult) {
        this.listenerResult = listenerResult
        list = ArrayList();
        adapter = ResultAdapter();
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater!!.inflate(R.layout.fragment_result, container, false)
        onClickResult(view)
        setupRecycler(view)
        return view
    }

    private fun setupRecycler(view : View) {
        tvResult = view.tvResult
        view.recyclerView!!.setHasFixedSize(true)
        view.recyclerView!!.layoutManager = LinearLayoutManager(activity,LinearLayoutManager.VERTICAL, false)
        view.recyclerView!!.addItemDecoration(com.michaldrabik.kotlintest.utilities.DividerItemDecoration(activity, LinearLayoutManager.VERTICAL, 8.dpToPx(activity)))
        view.recyclerView!!.adapter = adapter
    }

    override fun onResume() {
        super.onResume()
        adapter.clearItems();
        var bundle = Bundle()
        bundle = arguments
        val value = bundle.getString(Constant.TAG_VALUE)
        val result = bundle.getString(Constant.TAG_RESULT)
        if (result == "true") {
            tvResult!!.text = value + " : " + " is a palindrome"
        } else {
            tvResult!!.text = value + " : " + " is not a palindrome"
        }
        list!!.add(CPalindrome(value, result))
        adapter!!.addItems(list!!)
    }


    fun onClickResult(view : View) {
        view.btnResult.setOnClickListener({
            listenerResult!!.onResultAction()
        });
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    interface ListenerResult {
        fun onResultAction()
    }

}// Required empty public constructor
